Coverage in CI broken

Before posting the bug, take a look at the
[FAQ](https://github.com/eclipse-iceoryx/iceoryx2/blob/main/FAQ.md)
for a possible solution.

## Required information

**Operating system:**

Ubuntu

**Rust version:**
Rust 1.86, gcovr 8.5 

**iceoryx2 version:**
main branch

**Detailed log output:**
See https://github.com/eclipse-iceoryx/iceoryx2/actions/runs/20851745927/job/59910584597

```
Run ./internal/scripts/generate-cov-report.sh --html
(INFO) Reading coverage data...

(ERROR) Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/workers.py", line 96, in worker
    work(*args, **kwargs)
    ~~~~^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 531, in process_datafile
    done = run_gcov_and_process_files(
        abs_filename,
    ...<3 lines>...
        chdir=wd,
    )
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 946, in run_gcov_and_process_files
    process_gcov_text_data(
    ~~~~~~~~~~~~~~~~~~~~~~^
        gcov_filename, filename, covdata, options, chdir
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 326, in process_gcov_text_data
    covdata.insert_file_coverage(filecov, merge_mode)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/container.py", line 210, in insert_file_coverage
    self.data[key].merge(filecov, options)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 2353, in merge
    self.__functions.merge(other.__functions, options)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage_dict.py", line 57, in merge
    self[key].merge(item, options)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 2079, in merge
    self.raise_merge_error(
    ~~~~~~~~~~~~~~~~~~~~~~^
        f"Got function {self.name} on multiple lines: {', '.join([str(line) for line in lines])}.\n"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
        other,
        ^^^^^^
    )
    ^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 148, in raise_merge_error
    raise GcovrMergeAssertionError(
    ...<11 lines>...
    )
gcovr.exceptions.GcovrMergeAssertionError: /home/runner/work/iceoryx2/iceoryx2/iceoryx2-bb/cxx/include/iox2/legacy/detail/expected.inl:182 Got function <unknown function> on multiple lines: 182, 200.
	You can run gcovr with --merge-mode-functions=MERGE_MODE.
	The available values for MERGE_MODE are described in the documentation.
GCOV data file of merge source is:
   {('target/ff/cc/cmake_cov_build/component-tests/cxx/CMakeFiles/component-tests_cxx.dir/src/test_container_mutation.gcno', '/home/runner/work/iceoryx2/iceoryx2/expected.inl##f264698fde331b639607f2a356441a76.gcov')}
and of merge target is:
   {('target/ff/cc/cmake_cov_build/component-tests/cxx/CMakeFiles/component-tests_cxx.dir/src/main.gcno', '/home/runner/work/iceoryx2/iceoryx2/expected.inl##f264698fde331b639607f2a356441a76.gcov')}

Error: (ERROR) Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/workers.py", line 96, in worker
    work(*args, **kwargs)
    ~~~~^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 531, in process_datafile
    done = run_gcov_and_process_files(
        abs_filename,
    ...<3 lines>...
        chdir=wd,
    )
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 946, in run_gcov_and_process_files
    process_gcov_text_data(
    ~~~~~~~~~~~~~~~~~~~~~~^
        gcov_filename, filename, covdata, options, chdir
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/formats/gcov/read.py", line 326, in process_gcov_text_data
    covdata.insert_file_coverage(filecov, merge_mode)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/container.py", line 210, in insert_file_coverage
    self.data[key].merge(filecov, options)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 2353, in merge
    self.__functions.merge(other.__functions, options)
    ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage_dict.py", line 57, in merge
    self[key].merge(item, options)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 2079, in merge
    self.raise_merge_error(
    ~~~~~~~~~~~~~~~~~~~~~~^
        f"Got function {self.name} on multiple lines: {', '.join([str(line) for line in lines])}.\n"
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
        other,
        ^^^^^^
    )
    ^
  File "/opt/hostedtoolcache/Python/3.13.11/x64/lib/python3.13/site-packages/gcovr/data_model/coverage.py", line 148, in raise_merge_error
    raise GcovrMergeAssertionError(
    ...<11 lines>...
    )
gcovr.exceptions.GcovrMergeAssertionError: /home/runner/work/iceoryx2/iceoryx2/iceoryx2-bb/cxx/include/iox2/legacy/detail/expected.inl:182 Got function <unknown function> on multiple lines: 182, 200.
	You can run gcovr with --merge-mode-functions=MERGE_MODE.
	The available values for MERGE_MODE are described in the documentation.
GCOV data file of merge source is:
   {('target/ff/cc/cmake_cov_build/component-tests/cxx/CMakeFiles/component-tests_cxx.dir/src/test_container_mutation.gcno', '/home/runner/work/iceoryx2/iceoryx2/expected.inl##f264698fde331b639607f2a356441a76.gcov')}
and of merge target is:
   {('target/ff/cc/cmake_cov_build/component-tests/cxx/CMakeFiles/component-tests_cxx.dir/src/main.gcno', '/home/runner/work/iceoryx2/iceoryx2/expected.inl##f264698fde331b639607f2a356441a76.gcov')}

(ERROR) Error occurred while reading reports: Worker thread raised exception, workers canceled.
```

**Observed result or behaviour:**
With the recent release of gcovr 8.5 some changes where introduced that break the CI jobs.
https://gcovr.com/en/stable/changelog.html#january-2026

**Expected result or behaviour:**
Coverage job is green

**Conditions where it occurred / Performed steps:**
https://github.com/eclipse-iceoryx/iceoryx2/actions/runs/20851745927/job/59910584597

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -923,9 +923,9 @@ jobs:
           sudo groupadd testgroup2
 
       - name: Setup Rust
-        uses: dtolnay/rust-toolchain@888c2e1ea69ab0d4330cbf0af1ecc7b68f368cc1 # ratchet:dtolnay/rust-toolchain@v1
+        uses: dtolnay/rust-toolchain@e97e2d8cc328f1b50210efc529dca0028893a2d9 # ratchet:dtolnay/rust-toolchain@v1
         with:
-          toolchain: 1.86.0
+          toolchain: 1.92.0
           components: llvm-tools
 
       - name: Setup gcovr

